cli_tools/common/utils/daisyutils/workflow_hook.go

```diff
@@ -19,9 +19,18 @@ import (
 )
 
 // To rebuild the mock, run `go generate ./...`
-//go:generate go run github.com/golang/mock/mockgen -package mocks -source $GOFILE -destination ../../../mocks/mock_workflow_modifier.go
+//go:generate go run github.com/golang/mock/mockgen -package mocks -source $GOFILE -destination ../../../mocks/mock_workflow_hook.go
 
-// WorkflowModifier allows for modifying and validating a daisy workflow.
-type WorkflowModifier interface {
-	Modify(wf *daisy.Workflow) error
+// WorkflowPreHook executes before the workflow runs.
+type WorkflowPreHook interface {
+	// PreRunHook allows a WorkflowHook to modify a workflow prior to running.
+	PreRunHook(wf *daisy.Workflow) error
+}
+
```

Comment:
@zoran15 

> It seems all workflow hooks are preRun, and postRuns are no-op. Maybe consider having a single operation for each hook (e.g. just Run(w)) and register them as either pre- or post- run.

Good call! Updated to split the interface so that implementers don't have to create no-op functions.

Reply:
I think this looks good. Going a step further, have you considered having a single interface for hooks/modifiers (e.g. WorkflowHook) and same hook could be registered as either pre- or post- workflow hook?

Reply:
Yeah, the interface looks different enough that it doesn't make sense to merge them into one, as some args would be ignored in different scenarios.